### DISCUSSION TITLE
Remove segments located inside the grain

### DISCUSSION THREAD

**Discussion by Saoussen-ouhiba** [mtex-toolbox/mtex#683] [Ask Anything] [answered]
Hello,

I would like to remove the segments that are located inside the grain from the list of segments of the grain boundary. Below is attached an example of the segments that I want to get rid of (indicated by the red arrow).
![image](https://user-images.githubusercontent.com/73701277/103960275-0b149b00-5152-11eb-95a7-71c034a0a050.png)
Can someone help me to do this please? 
Thank you in advance for your help,

Saoussen

**Comment by ralfHielscher** [comment, marked as answer]
To ignore all inclusion grain boundaries you can do
```matlab
% all grain boundary segments
gB = grains.boundary;

gBnotIncl = gB(~any(ismember(gB.grainId,grains(grains.isInclusion).id),2)));

plot(gBnotIncl)
```
All the best,

Ralf.
